liquibase-core/src/main/java/liquibase/snapshot/SnapshotControl.java

```diff
@@ -16,10 +16,17 @@ public class SnapshotControl implements LiquibaseSerializable {
     private Set<Class<? extends DatabaseObject>> types;
     private SnapshotListener snapshotListener;
 
+    private Set<Class<? extends DatabaseObject>> excludedTypes;
+
     public SnapshotControl(Database database) {
         setTypes(DatabaseObjectFactory.getInstance().getStandardTypes(), database);
     }
 
+    public SnapshotControl(Database database, List<String> excludeList) {
```

Comment:
If we allow to pass null value as param in this constructor, let's:
1. Add `@Nullable` annotation
2. Inside of setExcludeTypes let's specify empty HashSet, so we can simplify logic inside of addType method from
```
    public boolean addType(Class<? extends DatabaseObject> type, Database database) {
        boolean added = false;
        if (excludedTypes == null || !excludedTypes.contains(type)) {
            added = this.types.add(type);
        }
        if (added) {
            for (Class<? extends DatabaseObject> container : SnapshotGeneratorFactory.getInstance().getContainerTypes(type, database)) {
                addType(container, database);
            }
        }

        return added;

    }
```

to

```
    public boolean addType(Class<? extends DatabaseObject> type, Database database) {
        if (excludedTypes.contains(type)) {
            return false;
        }
        
        boolean added = this.types.add(type);
        if (added) {
            for (Class<? extends DatabaseObject> container : SnapshotGeneratorFactory.getInstance().getContainerTypes(type, database)) {
                addType(container, database);
            }
        }
        
        return added;
    }
```

Reply:
@andrulban for 1.- in liquibase we don't have the required dependency to use `@Nullable` annotation, let me know if I should add it, if so, is there any special setting/configuration I should do for liquibase-bundle or somewhere else?

About 2.- Even if I add an empty HashSet in setExcludeTypes it may not be called before addType is used, so it will trigger a NPE. Right now, only the new constructor call setExcludeTypes. 

This will work only if we call setExcludeTypes from all other constructors. For that reason, I initialize the variable once it is defined. Let me know if this is ok for you

Reply:
HI @rsian-lb, sorry for so many comments, but I recalled that Andrii C always was strongly against new properties  in Liquibase fork, so although your solution will work, I would advice to remove new property and do all the magic inside of the new constructor, for me looks like it should also work, could you please verify this solution and if it works - then use it?

Reply:
@andrulban updated as requested.

Reply:
Hi @rsian-lb, looks better now, but it still overloads main Liquibase methods, I can see a way how we can change everything without any overloading of public methods, e.g.:

```
    private void setTypes(Set<Class<? extends DatabaseObject>> types, Database database, Set<Class<? extends DatabaseObject>> excludeTypes) {
        this.types = new HashSet<>();
        for (Class<? extends DatabaseObject> type : types) {
            if (!excludeTypes.contains(type)) {
                addType(type, database);
            }
        }
    }

    private void setTypes(Set<Class<? extends DatabaseObject>> types, Database database) {
        this.types = new HashSet<>();
        for (Class<? extends DatabaseObject> type : types) {
            addType(type, database);
        }
    }

    public boolean addType(Class<? extends DatabaseObject> type, Database database) {
        boolean added = this.types.add(type);
        if (added) {
            for (Class<? extends DatabaseObject> container : SnapshotGeneratorFactory.getInstance().getContainerTypes(type, database)) {
                addType(container, database);
            }
        }
        return added;
    }
```
Will this solution keep working same?